### ISSUE TITLE
MessageProps.$params wrong types

### ISSUE THREAD

**Issue by idcom-jakub-soboczynski** [vuelidate/vuelidate#1030]
Hi! 
when I want to display a personalized message with validator parameters, TypeScript throws me an error that the given key does not exist on the Object type.

`TS2339: Property 'min' does not exist on type 'object'.`

I use the validation I present below:

```typescript
minLength: helpers.withMessage(
      ({ $params }) => `The name should contain at least ${$params.min} chars.`,
      minLength(3),
    ),
```
I think there is a type problem in the `MessageProps` interface.

https://github.com/vuelidate/vuelidate/blob/2c3e82f29e3e1e1781bda3abf50ea6e355525cf9/packages/validators/index.d.ts#L78

Perhaps the solution would be to create a `MessagePropsParams` interface that would include all the arguments for the built-in validation, e.g.

```typescript
export interface MessagePropsParams {
  min: number;
  max: number;
  separator: string;
  etc...
}

export interface MessageProps {
  $model: string;
  $property: string;
  $params: MessagePropsParams;
  $validator: string;
  $pending: boolean,
  $invalid: boolean,
  $response: unknown,
  $propertyPath: string,
}
```

or maybe:
```typescript
export interface MessageProps {
  $model: string;
  $property: string;
  $params: { [attr: string]: any };
  $validator: string;
  $pending: boolean,
  $invalid: boolean,
  $response: unknown,
  $propertyPath: string,
}
```

**Comment by Riu** [comment]
Solved in #1037

**Comment by idcom-jakub-soboczynski** [comment]
Not really. It's not a problem with `ValidationRuleWithParams<{ length: number }>;`, just with `$params: object` in `MessageProps` interface, because it's the same problem with `maxLength` validator:
```typescript
TS2339: Property 'max' does not exist on type 'object'.
```

So i recommend to create interface for $params → `MessagePropsParams` and add it to `MessageProps` interface. Like in examples:

```typescript
export interface MessagePropsParams {
  min: number;
  max: number;
  separator: string;
  etc...
}

export interface MessageProps {
  $model: string;
  $property: string;
  $params: MessagePropsParams;
  $validator: string;
  $pending: boolean,
  $invalid: boolean,
  $response: unknown,
  $propertyPath: string,
}

or

export interface MessageProps {
  $model: string;
  $property: string;
  $params: { [attr: string]: any };
  $validator: string;
  $pending: boolean,
  $invalid: boolean,
  $response: unknown,
  $propertyPath: string,
}
```

I made a PR [#1033](https://github.com/vuelidate/vuelidate/pull/1033) where this problem is solved.

**Comment by dobromir-hristov** [comment]
Will merge it soon, just can't get to the oss world, sorry.

**Comment by dobromir-hristov** [comment]
Closed by #1033
